src/Hacks/antiaim.cpp

```diff
@@ -226,75 +91,77 @@ void AntiAim::CreateMove(CUserCmd* cmd)
+    CCSGOAnimState* animState = localplayer->GetAnimState();
 
-    if( Settings::AntiAim::LBYBreaker::enabled ){
-
-        if( vel2D >= 0.1f || !(localplayer->GetFlags() & FL_ONGROUND) || localplayer->GetFlags() & FL_FROZEN ){
+    if (Settings::AntiAim::LBYBreaker::enabled)
+    {
+        if (vel2D >= 0.1f || !(localplayer->GetFlags() & FL_ONGROUND) || localplayer->GetFlags() & FL_FROZEN)
+        {
             lbyBreak = false;
             lastCheck = globalVars->curtime;
-        } else {
-            if( !lbyBreak && ( globalVars->curtime - lastCheck ) > 0.22 ){
-                angle.y -= Settings::AntiAim::LBYBreaker::offset;
-                lbyBreak = true;
-                lastCheck = globalVars->curtime;
-                needToFlick = true;
-            } else if( lbyBreak && ( globalVars->curtime - lastCheck ) > 1.1 ){
-                angle.y -= Settings::AntiAim::LBYBreaker::offset;
-                lbyBreak = true;
-                lastCheck = globalVars->curtime;
-                needToFlick = true;
-            }
+            nextUpdate = globalVars->curtime + 0.22;
+        }
+        else if (!lbyBreak && (globalVars->curtime - lastCheck) > 0.22 || lbyBreak && (globalVars->curtime - lastCheck) > 1.1)
+        {
+            tempangle = Settings::AntiAim::LBYBreaker::offset;
+            lbyBreak = true;
+            lastCheck = globalVars->curtime;
+            nextUpdate = globalVars->curtime + 1.1;
+            needToFlick = true;
         }
     }
 
-    if (Settings::AntiAim::Yaw::enabled && !needToFlick)
-    {
-        DoAntiAimY(localplayer, angle, bSend);
+    if ((nextUpdate - globalVars->interval_per_tick) >= globalVars->curtime && nextUpdate <= globalVars->curtime)
+        CreateMove::sendPacket = false;
 
-        CreateMove::sendPacket = bSend;
-        if (Settings::AntiAim::HeadEdge::enabled && edging_head && !bSend)
-            angle.y = edge_angle.y;
-    }
+    static bool directionSwitch = false;
+
+    if (inputSystem->IsButtonDown(KEY_LEFT))
+		directionSwitch = true;
+	else if (inputSystem->IsButtonDown(KEY_RIGHT))
+		directionSwitch = false;
 
-    if (!ValveDSCheck::forceUT && (*csGameRules) && (*csGameRules)->IsValveDS())
+    if (needToFlick)
     {
-        if (Settings::AntiAim::Pitch::type >= AntiAimType_X::STATIC_UP_FAKE)
-            Settings::AntiAim::Pitch::type = AntiAimType_X::STATIC_UP;
+        CreateMove::sendPacket = false;
+        angle.y += tempangle;
     }
+    else
+    	DoAntiAim(angle, bSend, animState, directionSwitch);
 
-    if (Settings::AntiAim::Pitch::enabled)
-        DoAntiAimX(angle, bSend, should_clamp);
-
-    if( should_clamp ){
+    if (should_clamp)
+    {
         Math::NormalizeAngles(angle);
         Math::ClampAngles(angle);
     }
 
+    if (!needToFlick)
+        CreateMove::sendPacket = bSend;
+
+    if (bSend)
+        AntiAim::realAngle = CreateMove::lastTickViewAngles;
```

Comment:
Inaccurate method of getting real yaw

Reply:
Can you give me some suggestions of how to make it correct?